src/shared/sam/debugger/typescriptSamDebug.ts

```diff
@@ -113,44 +113,72 @@ export async function makeTypescriptConfig(config: SamLaunchRequestArgs): Promis
 }
 
 /**
- * For non-template debug configs (target = code), compile the project
- * using a temporary default tsconfig.json file.
+ * Compiles non-template (target=code) debug configs, using a temporary default
+ * tsconfig.json file.
+ *
+ * Assumes that `sam build` was already performed.
  */
 async function compileTypeScript(config: NodejsDebugConfiguration): Promise<void> {
-    if (config.invokeTarget.target === 'code') {
-        const tscResponse = await new ChildProcess(true, 'tsc', undefined, '-v').run()
-        if (tscResponse.exitCode !== 0 || !tscResponse.stdout.startsWith('Version')) {
-            throw new Error('TypeScript compiler "tsc" not found.')
-        }
-        const samBuildOutputAppRoot = path.join(
-            config.baseBuildDir!,
-            'output',
-            path.parse(config.invokeTarget.projectRoot).name
-        )
-        const tsconfigPath = path.join(samBuildOutputAppRoot, 'tsconfig.json')
-        if (
-            (await readdir(config.codeRoot)).includes('tsconfig.json') ||
-            (await hasFileWithSuffix(config.codeRoot, '.ts', '**/node_modules/**'))
-        ) {
-            //  This default config is a modified version from the AWS Toolkit for JetBrain's tsconfig file. https://github.com/aws/aws-toolkit-jetbrains/blob/911c54252d6a4271ee6cacf0ea1023506c4b504a/jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/lambda/nodejs/NodeJsLambdaBuilder.kt#L60
-            const defaultTsconfig = {
-                compilerOptions: {
-                    target: 'es6',
-                    module: 'commonjs',
-                    typeRoots: ['node_modules/@types'],
-                    types: ['node'],
-                    rootDir: '.',
-                    inlineSourceMap: true,
-                },
-            }
-            try {
-                writeFileSync(tsconfigPath, JSON.stringify(defaultTsconfig, undefined, 4))
-                getLogger('channel').info('Compiling TypeScript')
-                await new ChildProcess(true, 'tsc', undefined, '--project', samBuildOutputAppRoot).run()
-            } catch (error) {
-                getLogger('channel').error(`Compile Error: ${error}`)
-                throw Error('Failed to compile typescript Lambda')
-            }
+    if (!config.baseBuildDir) {
+        throw Error('invalid state: config.baseBuildDir was not set')
+    }
+    if (config.invokeTarget.target !== 'code') {
+        return
+    }
+
+    async function findTsOrTsConfig(dir: string, child: boolean): Promise<string | undefined> {
```

Comment:
I'm not huge on nested named functions unless there's a reason to have it nested (e.g. using context directly from the outer function without params). Any downside to pulling this out?

Reply:
My reply is the same every time this is raised. Local functions are like local variables. They are not special and they should be allowed, to indicate the ad-hoc/throwaway nature of the function.

Reply:
Locally scoped functions are only problematic for performance-critical code. Other than that, local functions are great since:
* They are located closest to where they're used
* They capture the current scope (no need to pass around a bunch of variables)
* Avoids the need to generalize a function that really shouldn't be generalized